python/ray/data/_internal/plan.py

```diff
@@ -452,11 +452,6 @@ def _get_source_blocks_and_stages(
             # beginning.
             blocks = self._in_blocks
             stats = self._in_stats
-            if not self.has_lazy_input():
-                # If not a lazy datasource, unlink the input blocks from the plan so we
-                # can eagerly reclaim the input block memory after the first stage is
-                # done executing.
-                self._in_blocks = None
```

Comment:
I think that removing this is safe, but when adding the block clearing logic for the new execution plan/model, we should be sure to have comprehensive memory releasing tests that covers all corners cases of our block clearing semantics.

Reply:
@clarkzinzow - agreed.